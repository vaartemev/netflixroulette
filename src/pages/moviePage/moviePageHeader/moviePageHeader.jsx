import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { withRouter, useRouter } from 'next/router';
import Link from 'next/link';
import { MovieDetails } from './movieDetails';
import { Logo, Button, ButtonType } from '../../../components';

import { getMovieDetailsById } from '../../../actions';
import './moviePageHeader.scss';

const mapStateToProps = state => ({
  movie: state.movie.get('movie'),
});

const mapDispatchToProps = dispatch => ({
  getMovieDetails: id => {
    dispatch(getMovieDetailsById(id));
  },
});

export const MoviePageHeader = withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps,
  )(({ movie, getMovieDetails }) => {
    const router = useRouter();
    const { id } = router.query;
    useEffect(() => {
      getMovieDetails(id);
    }, []);
    return (
      <div className="header">
        <Link href="localhost:3000/search">
          <li>aaaaaa</li>
        </Link>
        <Logo text="netflixroulette" />
        {Object.keys(movie).length && (
          <>
            <Link href="/">
              <Button type={ButtonType.searchLink} text="Search" />
            </Link>
            <MovieDetails movie={movie} />
          </>
        )}
      </div>
    );
  }),
);
