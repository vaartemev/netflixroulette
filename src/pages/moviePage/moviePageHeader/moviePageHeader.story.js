import React from 'react';
import { storiesOf } from '@storybook/react';
import { MoviePageHeader } from './moviePageHeader';

storiesOf('Header with movie', module).add('Header with movie', () => (
  <MoviePageHeader />
));
