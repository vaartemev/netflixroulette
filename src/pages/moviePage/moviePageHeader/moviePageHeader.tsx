import React, { useEffect } from 'react';
import { useRouter, withRouter } from 'next/router';
import { useSelector, useDispatch } from 'react-redux';
import { MovieDetails } from './movieDetails';
import { Logo, Button, ButtonType, Link } from '../../../components';

import { getMovieDetailsById } from '../../../actions';
import { movieSelector } from '../../../selectors';
import './moviePageHeader.scss';

export const MoviePageHeader = withRouter(() => {
  const movie = useSelector(movieSelector);
  const dispatch = useDispatch();

  const router = useRouter();
  const { id } = router.query;
  useEffect(() => {
    dispatch(getMovieDetailsById(Number(id)));
  }, [id]);

  return (
    <div className="header">
      <Logo text="netflixroulette" />

      {Object.keys(movie).length && (
        <>
          <Link href="/search">
            <Button type={ButtonType.searchLink} text="Search" />
          </Link>
          <MovieDetails movie={movie} />
        </>
      )}
    </div>
  );
});
