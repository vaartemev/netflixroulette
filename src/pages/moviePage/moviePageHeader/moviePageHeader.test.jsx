import React from 'react';
import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { MoviePageHeader } from './moviePageHeader';
import { configureStore } from '../../../store';

const store = configureStore();

describe('Movie page header', () => {
  const output = mount(
    <Router>
      <Provider store={store}>
        <MoviePageHeader />
      </Provider>
    </Router>,
  );
  it('Render correctly', () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
