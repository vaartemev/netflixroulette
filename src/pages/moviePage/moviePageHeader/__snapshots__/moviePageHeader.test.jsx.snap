// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie page header Render correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <withRouter(Connect(Component))>
        <Connect(Component)
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <Component
            getMovieDetails={[Function]}
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            movie={Object {}}
          >
            <div
              className="header"
            >
              <Logo
                text="netflixroulette"
              >
                <div
                  className="logo"
                >
                  netflixroulette
                </div>
              </Logo>
              0
            </div>
          </Component>
        </Connect(Component)>
      </withRouter(Connect(Component))>
    </Provider>
  </Router>
</BrowserRouter>
`;
